drivers/wifi/siwx917/siwx917_wifi.c

```diff
@@ -311,6 +312,107 @@ static void siwx917_ethernet_init(struct net_if *iface)
 	}
 }
 
+#ifdef CONFIG_WIFI_SIWX917_AP_MODE
+static int siwx917_ap_enable(const struct device *dev, struct wifi_connect_req_params *params)
+{
+	struct siwx917_dev *sidev = dev->data;
+	sl_wifi_ap_configuration_t configuration;
+	int ret;
+
+	__ASSERT(params, "params should not be NULL");
+
+	configuration.channel.channel = params->channel;
+	configuration.encryption = SL_WIFI_CCMP_ENCRYPTION;
+	configuration.rate_protocol = SL_WIFI_RATE_PROTOCOL_AUTO;
+	configuration.options = 0;
+	configuration.credential_id = SL_NET_DEFAULT_WIFI_AP_CREDENTIAL_ID;
+	configuration.keepalive_type = SL_SI91X_AP_NULL_BASED_KEEP_ALIVE;
+	configuration.beacon_interval = CONFIG_WIFI_SIWX917_BEACON_INTERVAL;
+	configuration.client_idle_timeout = 0xFF;
+	configuration.dtim_beacon_count = CONFIG_WIFI_SIWX917_DTIM_BEACON_COUNT;
+	configuration.maximum_clients = CONFIG_WIFI_SIWX917_MAX_STATIONS;
+	configuration.beacon_stop = 0;
+	configuration.tdi_flags = SL_WIFI_TDI_NONE;
+	configuration.is_11n_enabled = 1;
+	configuration.ssid.length = params->ssid_length;
```

Comment:
If new member are added to `sl_wifi_ap_configuration_t`, this code won't initialize them. It is better to use:
```
sl_wifi_ap_configuration_t configuration = { };
```
You can take this opportunity to initialize some members:
```
sl_wifi_ap_configuration_t configuration = {
	.channel.channel = params->channel,
	.encryption = SL_WIFI_CCMP_ENCRYPTION,
	.channel.band = SL_WIFI_BAND_2_4GHZ,
	...
};
```
It is not required to initialize `beacon_stop` or `options` since they are already 0.

Compile will complain `__ASSERT()` is called after access to `params`. Personally, I would use `params` without check (it's bug in the caller). However, it is up to you.